tests/generated_data/generate_md.py

```diff
@@ -0,0 +1,153 @@
+        "keyval": {
+            "public": "250f9ae3d1d3d5c419a73cfb4a470c01de1d5d3d61a3825416b5f5d6b88f4a30",
+            "private": "e6645b00312c8a257782e3e61e85bafda4317ad072c52251ef933d480c387abd",
+        },
+    },
+    "ed25519_3": {
+        "keytype": "ed25519",
+        "scheme": "ed25519",
+        "keyid": "3458204ed467519c19a5316eb278b5608472a1bbf15850ebfb462d5315e4f86d",
+        "keyval": {
+            "public": "82380623abb9666d4bf274b1a02577469445a972e5650d270101faa5107b19c8",
+            "private": "cd13dd2180334b24c19b32aaf27f7e375a614d7ba0777220d5c2290bb2f9b868",
+        },
+    },
+    "ed25519_4": {
+        "keytype": "ed25519",
+        "scheme": "ed25519",
+        "keyid": "2be5c21e3614f9f178fb49c4a34d0c18ffac30abd14ced917c60a52c8d8094b7",
+        "keyval": {
+            "public": "0e6738fc1ac6fb4de680b4be99ecbcd99b030f3963f291277eef67bb9bd123e9",
+            "private": "7e2e751145d1b22f6e40d4ba2aa47158207acfd3c003f1cbd5a08141dfc22a15",
+        },
+    },
+}
+
+keys: Dict[str, Key] = {
+    "ed25519_1": Key.from_securesystemslib_key(keys_dict["ed25519_1"]),
+    "ed25519_2": Key.from_securesystemslib_key(keys_dict["ed25519_2"]),
+    "ed25519_3": Key.from_securesystemslib_key(keys_dict["ed25519_3"]),
+    "ed25519_4": Key.from_securesystemslib_key(keys_dict["ed25519_4"]),
+}
+
+
+class MetadataGenerator:
+    """Responsible for static metadata files generation.
+
+    Args:
+        out_dir: The output dir for the newly generated metadata files.
+    """
+
+    def __init__(self, out_dir: Optional[str] = None) -> None:
+        expires_str = "2050-01-01T00:00:00Z"
+        self.expiry = datetime.strptime(expires_str, "%Y-%m-%dT%H:%M:%SZ")
+        self.dir = out_dir if out_dir else "ed25519_metadata"
+        if not os.path.exists(self.dir):
+            os.mkdir(self.dir)
+
+        self.spec_version = ".".join(SPECIFICATION_VERSION)
+        # Dump metadata files in a human readable format.
+        self.serializer = JSONSerializer()
+
+    def verify_generation(self, md: Metadata, path: str) -> None:
+        """Verify that newly generated file equals the locally stored one.
+
+        Args:
+            md: Newly generated metadata object.
+            path: Path to the locally stored metadata file.
+        """
+        with open(path, "rb") as f:
+            static_md_bytes = f.read()
+            md_bytes = md.to_bytes(self.serializer)
+            if static_md_bytes != md_bytes:
+                raise ValueError(
+                    f"Generated data != local data at {path}. Generate a new "
+                    + "metadata with 'python generated_data/generate_md.py'"
+                )
+
+    def generate_all_files(
+        self, dump: Optional[bool] = False, verify: Optional[bool] = False
+    ) -> None:
+        """Generate a new repository and optionally verify it.
+
+        Args:
+            dump: Wheter to dump the newly generated files inside self.dir.
```

Comment:
Is there a use case where the files are not dumped?

Reply:
Yes, when we are testing that functionality https://github.com/MVrachev/tuf/blob/45f20beaa83bd2b8b6dbd44164e1a1b7883bd741/tests/test_metadata_generation.py#L20.